src/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -188,30 +187,40 @@ internal DateTimeOffset GetLastWriteTime(ReadOnlySpan<char> path, bool continueO
             return UnixTimeToDateTimeOffset(_fileStatus.MTime, _fileStatus.MTimeNsec);
         }
 
-        internal void SetLastWriteTime(string path, DateTimeOffset time)
-            => SetAccessWriteTimes(path, null, time.ToUnixTimeSeconds());
-
+        internal void SetLastWriteTime(string path, DateTimeOffset time) => SetAccessWriteTimes(path, null, time);
+        
         private DateTimeOffset UnixTimeToDateTimeOffset(long seconds, long nanoseconds)
         {
             return DateTimeOffset.FromUnixTimeSeconds(seconds).AddTicks(nanoseconds / NanosecondsPerTick).ToLocalTime();
         }
 
-        private void SetAccessWriteTimes(string path, long? accessTime, long? writeTime)
+        private void SetAccessWriteTimes(string path, DateTimeOffset? accessTime, DateTimeOffset? writeTime)
         {
             // force a refresh so that we have an up-to-date times for values not being overwritten
             _fileStatusInitialized = -1;
             EnsureStatInitialized(path);
 
+            long? accessTimeSeconds = accessTime?.ToUnixTimeSeconds();
```

Comment:
You'd make this even clearer if you keep the times together.

``` c#
if (accessTime.HasValue)
{
    long seconds = accessTime.ToUnixTimeSeconds();
    buf[0].TvSec = seconds;
    buf[0].TvNsec = (accessTime.ToUnixTimeMilliseconds() - seconds * 1000) * 1_000_000;
    buf[1].TvSec = _fileStatus.MTime;
    buf[1].TvNsec = _fileStatus.MTimeNsec;
}

// etc.
```

Reply:
If it concerns you that both might come in, change the method to take `DateTimeOffset time, bool isAccessTime`.